tests/Fixtures/Integration/misc/PHP8_2.test

```diff
@@ -0,0 +1,179 @@
+--TEST--
+PHP 8.2 test.
+--RULESET--
+{
+    "@PhpCsFixer": true,
+    "@PHP82Migration": true
+}
+--REQUIREMENTS--
+{"php": 80200}
+--EXPECT--
+<?php
+
+// https://wiki.php.net/rfc/readonly_classes
+readonly final class Foo
+{
+    public string $c;
+    public string $prop;
+    public string $b;
+}
+
+// https://wiki.php.net/rfc/null-false-standalone-types
+class FalseNull
+{
+    public function __construct(
+        private false $falsy = false,
+        private null $nully = null
+    ) {
+    }
+
+    public function falsy(): false
+    {
+        return $this->falsy;
+    }
+
+    public function nully(): null
+    {
+        return $this->nully;
+    }
+}
+
+// https://wiki.php.net/rfc/true-type
+class TrueType
+{
+    public function __construct(
+        private true $havingABud = true
+    ) {
+    }
+
+    public function havingABud(): true
+    {
+        return $this->havingABud;
+    }
+}
+
+// https://wiki.php.net/rfc/dnf_types
+interface DnfA
+{
+    public function foo(): void;
+
+    public function bar(): void;
+}
+interface DnfB
+{
+    public function foo(): void;
+}
+interface DnfC
+{
+    public function bar(): void;
+}
+
+class DnfTypes
+{
+    public function __construct(
+        private DnfA|(DnfB&DnfC) $notSure
+    ) {
+    }
+
+    public function notSure(): DnfA|(DnfB&DnfC)
```

Comment:
I don't know if it's desired formatting 🤔 But it's close to what Fixer does right now (output is `DnfA|(DnfB·&DnfC)`), so I think it's OK.

Reply:
Formatting (spacing) should be probably configurable for DNF (no spaces by default?), but there is [bigger problem](https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/6608#issuecomment-1237503273) for now. Let's close this discussion and focus on fixing complex DNF support.